risc0/zkvm/src/receipt.rs

```diff
@@ -173,45 +170,141 @@ pub struct SessionReceipt {
     pub journal: Vec<u8>,
 }
 
-/// Provide common functionality implemented by all receipt types.
-#[dyn_partial_eq]
-#[typetag::serde(tag = "type")]
-pub trait Receipt: Debug {
+/// An inner receipt can take the form of a collection of [SegmentReceipts] or a
+/// [SuccinctReceipt].
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub enum InnerReceipt {
+    /// The [SegmentReceipts].
+    Flat(SegmentReceipts),
+
+    /// The [SuccinctReceipt].
+    Succinct(SuccinctReceipt),
+
+    /// A fake receipt for testing and development.
```

Comment:
Adding this suggested comment since it's the first question I had when I saw this enum variant.
```suggestion
    /// A fake receipt for testing and development.
    /// Verification will always return [VerificationError::InvalidProof]
```

Reply:
This will actually form the basis for "dev mode", and in that mode it won't return InvalidProof.